crates/evm/core/src/opts.rs

```diff
@@ -102,7 +102,8 @@ impl EvmOpts {
         )
         .await
         .wrap_err_with(|| {
-            format!("Could not instantiate forked environment with fork url: {fork_url}")
+            let truncated_url: String = fork_url.to_string().drain(..fork_url.len() / 2).collect();
+            format!("Could not instantiate forked environment with fork url: {truncated_url}[..]")
```

Comment:
Tested with both type of API keys and URLs and it works but a bit unclear from reading the code, mind adding a comment here?

Alternative would be to regex on the last `/`

Reply:
yeah, that's to avoid printing the entire URL in case of failure, for example changing a char in URL / API key would trigger such. I opted for half truncating as there could be URLs accepting key as a query param, so regex on the last `/` won't work

Reply:
Yeah I think `Url::parse`  would work best here